sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressSelector.java

```diff
@@ -61,7 +62,7 @@ public static Uri getPrimaryUri(RxDocumentServiceRequest request, List<AddressIn
             // Primary endpoint (of the desired protocol) was not found.
             throw new GoneException(String.format("The requested resource is no longer available at the server. Returned addresses are {%s}",
                                                   String.join(",", replicaAddresses.stream()
-                                                      .map(address -> address.getPhysicalUri().getURIAsString()).collect(Collectors.toList()))), null);
+                                                      .map(address -> address.getPhysicalUri().getURIAsString()).collect(Collectors.toList()))), null, HttpConstants.SubStatusCodes.UNKNOWN);
```

Comment:
nit: there is a method public GoneException(String msg) {} which will automatically populate UNKNOWN status code, you do not need to explicitly populate it

Reply:
Fixed the code